shared/utils/domains.ts

```diff
@@ -1,69 +1,53 @@
+import invariant from "invariant";
 import { trim } from "lodash";
 
 type Domain = {
-  tld: string;
   subdomain: string;
   domain: string;
+  custom?: boolean;
 };
 
+// strips protocol and whitespace from input
+// then strips the path and query string
+function normalizeUrl(url: string) {
+  return trim(url.replace(/(https?:)?\/\//, "")).split(/[/:?]/)[0];
+}
+
 // we originally used the parse-domain npm module however this includes
 // a large list of possible TLD's which increase the size of the bundle
 // unnecessarily for our usecase of trusted input.
+
+// TODO: make this private
+// this only applies to internal domains (hosted domains)
 export function parseDomain(url?: string): Domain | null | undefined {
```

Comment:
A lot of complexity in handing and using `parseDomain` comes from the optional argument for `url` — do you know what situations we would run into where url would come in as `undefined`?

Reply:
I'm not aware of anywhere that matters